CMakeLists.txt

```diff
@@ -149,6 +149,9 @@ set(ABSL_LOCAL_GOOGLETEST_DIR "/usr/src/googletest" CACHE PATH
   "If ABSL_USE_GOOGLETEST_HEAD is OFF and ABSL_GOOGLETEST_URL is not set, specifies the directory of a local GoogleTest checkout."
   )
 
+option(ABSL_USE_HARDENED_MODE
+  "If ON, abseill will sue hardened mode." OFF)
```

Comment:
Please fix the spelling here.

Reply:
Fixed the spelling in commit 43795ba.

Reply:
Added hardened mode to cmake test builds where I found that shared mode is tested in commit d767e1e.

I have checked the changes against typos but I can't test it so if you can test it in your environment please test it and propose changes if I did something wrong.

Or is there a way how I can test the testing build scripts on my own?

Reply:
So maybe I should be able to test it on my Ubuntu 20.04 and my macOS Ventura.

Reply:
linux_gcc_alpine_cmake.sh and linux_gcc-latest_libstdcxx_cmake.sh and macos_xcode_cmake.sh finished successfully on my systems (Ubuntu Focal 20.04 and macOS Ventura 13.6.4)

Reply:
I can't test windows_msvc_cmake.bat so please run tests in your env if possible.

If you would like to test the new cmake option somehow else please let me know.

Another approach would be set 0 to hardened mode in absl_alternate_options.h and set -DABSL_USE_HARDENED_MODE=ON in the cmake test scripts but I didn't find where the absl_alternate_options.h is used :/